### PR TITLE
Support ApiException.Content when actual type is ValidationApiException

### PR DESCRIPTION
**Pull request by DrakeLambert**
Fixes reactiveui/refit#636

Consider this usage of a Refit client:
```csharp
try
{
    await refitClient.GetAsync(123);
}
catch (ApiException e)
{
    Console.WriteLine(e.Content);
}
```

If the response follows [RFC 7808](https://tools.ietf.org/html/rfc7807), then `e.Content` will unexpectedly be `null`.

This PR would change the behavior so that `e.Content` would still be set to the string content of the response message. If the exception is caught as `ValidationApiException`, the `public ProblemDetails Content { get; }` will still function as normal.

**What kind of change does this PR introduce?**

Enhancement: Support use of `ApiException.Content` when hidden by `ValidationApiException.Content`.

**What is the current behavior?**

`ApiException.Content` is left as default, `null`, when hidden by `ValidationApiException.Content`.

**What is the new behavior?**

`ApiException.Content` is set to the string content of the response message.

**What might this PR break?**

If consumers rely on `ApiException.Content` to be `null` when the actual type is `ValidationApiException`, then their code may break. As far as I can tell, the alternative is more of an issue than this fix.

**Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

What docs should be added for this? I neglected this step because I considered the original behavior to be unexpected.

### CODE REVIEW

**Comment by jamiehowarth0** [comment]
Brilliant, thanks for this! I'll review shortly

**Comment by DrakeLambert** [comment]
@jamiehowarth0 who else could give a review here? I'd like to see what y'all think of this change.

**Comment by jamiehowarth0** [comment]
> @jamiehowarth0 who else could give a review here? I'd like to see what y'all think of this change.

Sorry for the delay, I'll check this in the next 24hrs

**Comment by DrakeLambert** [comment]
@jamiehowarth0 ping!

File: Refit.Tests/ResponseTests.cs [hunk 1]
```diff
@@ -221,6 +221,31 @@ public async Task BadRequestWithEmptyContent_ShouldReturnApiException()
...
+            var expectedResponse = new HttpResponseMessage(HttpStatusCode.BadRequest)
+            {
+                Content = new StringContent(expectedContent)
+            };
+            expectedResponse.Content.Headers.ContentType = new MediaTypeHeaderValue("application/problem+json");
+            mockHandler.Expect(HttpMethod.Get, "http://api/aliasTest")
+                .Respond(req => expectedResponse);
+
+            var actualException = await Assert.ThrowsAsync<ValidationApiException>(() => fixture.GetTestObject());
+            var actualBaseExcpetion = actualException as ApiException;
```
Comment (jamiehowarth0) [inline Refit.Tests/ResponseTests.cs:246]: Please can you rename these (typo in "exception")? I'll accept the PR

**Review by jamiehowarth0 — CHANGES_REQUESTED** [review]

File: Refit.Tests/ResponseTests.cs (hunk 1 above)
Comment (DrakeLambert) [reply Refit.Tests/ResponseTests.cs]: Done!

**Pull request merged**